feat(docs): Outlines the use of cryptography in Pro for WSL (#1454)
canonical/ubuntu-pro-for-wsl: docs/explanation/security-overview.md

> [Intune configuration options for WSL](https://learn.microsoft.com/en-us/windows/wsl/intune?source=recommendations)

### Cryptography used by Ubuntu Pro for WSL

> Ubuntu Pro for WSL use of cryptography starts at installation time: an MSIX package can only be

Reviewer: ```suggestion
The use of cryptography in Ubuntu Pro for WSL starts at installation time: an MSIX package can only be
```

> installed on Windows if its signed by a trusted source. Even when downloading the MSIX from GitHub

Reviewer: ```suggestion
installed on Windows if it's signed by a trusted source. Even when downloading the MSIX from GitHub
```

releases or from Canonical website, the binary package is signed by Microsoft Store, thus no

> security exceptions are required to allow installing the application.

Reviewer: ```suggestion
security exceptions are required when installing the application.
```

At runtime the Windows background agent communicates with the wsl-pro-service within the Ubuntu on
WSL instances via mutual TLS. Even though the network is local, virtual and private, data in flux is
always encrypted. During startup the agent creates self-signed certificates (removing previously

> existing ones) which are used to ensure data is encrypted both ways.

Reviewer: ```suggestion
always encrypted. During startup, the agent creates self-signed certificates (removing previously
existing ones) that are used to ensure data is encrypted in both directions.
```

I think "in both directions" might be more clear here, as "way" can sometimes be used in the sense of _direction_ or _method_.

> Those features are not configurable by the user.

Reviewer: ```suggestion
These cryptographic features are not configurable by the user.
```

Both the Windows agent and the WSL instances use TLS when communicating with a Landscape server, 
relying on the operating system's certificate store by default. It's possible to configure a connection
to use a custom certificate issued for that server or even plain TCP, although that's
discouraged for security reasons and unlikely to be useful in real scenarios, other than local

> testing. For more information refer to [the Landscape configuration reference page](howto::config-landscape).

Reviewer: It probably wouldn't create too many problems if we kept it, but I think removing it is probably the right call, especially because we don't want to create any unnecessary confusion in a security section.

All cryptographic functionalities used by Ubuntu Pro for WSL are provided by the Go standard

> library.

Reviewer: Or a specific package in the standard library?

### Decommissioning

To support remote configuration and for sharing of certain files between the system components, Pro
for WSL relies on non-virtualised access to the Windows registry `HKCU` hive and the user's profile

> directory. For that reason just uninstalling the MSIX may leave behind unwanted data in both

Reviewer: ```suggestion
directory. For that reason, uninstalling the MSIX may leave behind unwanted data in both
```
